app/services/policies/policies_list_service.rb

```diff
@@ -6,12 +6,7 @@
 class Policies::PoliciesListService
   def self.call(account, builtin: true)
     list = PolicyList.new
-    if builtin
-      response = ::JSONClient.get(ThreeScale.config.sandbox_proxy.apicast_registry_url)
-      return unless response.ok?
-
-      list.merge!(PolicyList.from_hash(response.body['policies']))
-    end
+    list.merge!(PolicyList.from_hash(fetch_policies_from_apicast)) if builtin
```

Comment:
So when the fetch fails, we still display custom policies?
I think we should raise a custom error then, and catch it where it is used.

So that it is taken care case by case

Reply:
I thought that would be done in another issue: [THREESCALE-2312 - Show feedback in integration settings UI when policies not available](https://issues.jboss.org/browse/THREESCALE-2312) 🤔 
Although to me they both should be done in the same one, because it is not an urgent task and it is better done together, but as I said in a previous comment, as long as it is done, I don't really mind much when/where 😄

Reply:
I mean at least keep the behaviour. Do not display the policies. if there is an error fetching them or if the URL is not present

Reply:
And in fact `!merge(nil)` will raise an error anyway and it is not tested :p

Reply:
> And in fact !merge(nil) will raise an error anyway and it is not tested :p

It won't: https://github.com/3scale/porta/pull/736#discussion_r274835846

> I mean at least keep the behaviour. Do not display the policies. if there is an error fetching them or if the URL is not present

right, keeping the same behaviour makes sense 👍

Reply:
Right! Sorry! 🙇